yellowbrick/classifier.py

```diff
@@ -50,6 +51,149 @@ def __init__(self, model, ax=None, **kwargs):
 
         super(ClassificationScoreVisualizer, self).__init__(model, ax=ax, **kwargs)
 
+##########################################################################
+## ConfusionMatrix
+##########################################################################
+
+class ConfusionMatrix(ClassificationScoreVisualizer):
+    """
+    Creates a heatmap visualization of the sklearn.metrics.confusion_matrix().
+    Initialization: Requires a classification model
+    """
+    def __init__(self, model, ax=None, classes=None, **kwargs):
+        """
+        Provide a classifier model
+
+        Parameters
+        ----------
+        :param model: the Scikit-Learn estimator
+            Should be an instance of a classifier, else the __init__ will
+            return an error.
+
+        :param ax: the matplotlib axis to plot the figure on (if None, a new axis will be created)
+
+        :param classes: a list of class names to use in the confusion_matrix.
+            This is passed to the 'labels' parameter of sklearn.metrics.confusion_matrix(), and follows the behaviour
+            indicated by that function. It may be used to reorder or select a subset of labels.
+            If None, values that appear at least once in y_true or y_pred are used in sorted order.
+            Default: None
+        """
+        super(ConfusionMatrix, self).__init__(model, ax=ax, **kwargs)
+        #Parameters provided by super (for reference during development only):
+        #self.ax
+        #self.size
+        #self.color
+        #self.title
+        #self.estimator
+        #self.name
+
+        #Initialize all the other attributes we'll use (for coder clarity)
+        self.confusion_matrix = None
+
+        self.cmap = color_sequence(kwargs.pop('cmap', 'YlOrRd'))
+
+        #If possible, assign the classes. If this can't happen now, it will happen in the .fit() method
```

Comment:
This is absolutely one of the things that needs to be done; e.g. what if a fitted model is passed into the classifier, then `fit()` won't be called. The question then becomes; what happens if the `estimator.classes_` is not None, but then `fit()` is called? Right now, if the user specifies the classes, then we don't want `fit()` to override the user-specified classes, but if we get them off the estimator, then we _do_ want to change the classes on fit. 

So here's my suggestion (that I think we'll use in almost all the classifier score models); let's make `Visualizer.classes` a property rather than a simple attribute that performs the lookup on demand: 

```python
def __init__(self, model, ax=None, classes=None, **kwargs):
    self._classes = classes 

@property
def classes(self):
    if self._classes is None: 
        try:
            return self.estimator.classes_ 
        except AttributeError:
            return None 
    return self._classes 

@classes.setter
def classes(self, value):
    self._classes = value 
```

Then in `fit()` we can check the value of `self._classes` but in `draw()` we can simply use self.classes. Thoughts?